tools-and-tests/tools/src/main/java/org/hiero/block/tools/days/subcommands/GcpLsCommand.java

```diff
@@ -0,0 +1,374 @@
+            if (r.gcpSigCount < 0) {
+                status = "ERROR";
+            }
+            System.out.printf(
+                    "%-8s %-10d %-10d %-10d %-8s%n", "0.0." + r.node, r.gcpSigCount, r.localSigCount, r.diff, status);
+        }
+
+        printDifferenceView(results);
+        printLsView(results);
+    }
+
+    private void printDifferenceView(List<NodeResult> results) {
+        // enriched differences view
+        if (!skipLocalCompare) {
+            for (final NodeResult r : results) {
+                if ((r.gcpOnly != null && !r.gcpOnly.isEmpty()) || (r.localOnly != null && !r.localOnly.isEmpty())) {
+                    System.out.printf("    Differences for node 0.0.%d:%n", r.node);
+                    if (r.gcpOnly != null && !r.gcpOnly.isEmpty()) {
+                        System.out.println("      Present in GCP only:");
+                        r.gcpOnly.forEach(f -> System.out.println("        " + f));
+                    }
+                    if (r.localOnly != null && !r.localOnly.isEmpty()) {
+                        System.out.println("      Present in local only:");
+                        r.localOnly.forEach(f -> System.out.println("        " + f));
+                    }
+                }
+            }
+        }
+    }
+
+    private void printLsView(List<NodeResult> results) {
+        // optional full ls-style view
+        if (showLs) {
+            for (final NodeResult r : results) {
+                System.out.printf("    LS for node 0.0.%d:%n", r.node);
+
+                if (r.gcpFiles != null && !r.gcpFiles.isEmpty()) {
+                    System.out.println("      GCP files:");
+                    r.gcpFiles.forEach(f -> System.out.println("        " + f));
+                } else {
+                    System.out.println("      GCP files: (none)");
+                }
+
+                if (!skipLocalCompare) {
+                    if (r.localFiles != null && !r.localFiles.isEmpty()) {
+                        System.out.println("      Local files:");
+                        r.localFiles.forEach(f -> System.out.println("        " + f));
+                    } else {
+                        System.out.println("      Local files: (none)");
+                    }
+                }
+            }
+        }
+    }
+
+    private void validateInputs() {
+        if (endDate.isBefore(startDate)) {
+            throw new IllegalArgumentException("end-date must be >= start-date");
+        }
+    }
+
+    /**
+     * Check one node for a given date+hour:
+     *
+     *  - GCP: uses MainNetBucket.listHour(date),
+     *         then filters blobs for this node and record file extensions.
+     *  - Local: counts *.rcd / *.rcd.gz files in downloadedDaysDir for date+hour+node.
+     */
+    private NodeResult checkNodeForHour(LocalDate date, int hour, int node) {
+        List<String> gcpSigFiles = List.of();
+        long gcpSigCount;
```

Comment:
For clarity I think you should default the counts and diff to 0, then set them to -1 on failures or skips (skips should be clear as the printouts should input)
Then in the else that sets `localRecordFiles` below it should be an `if else` conditions.
In this way you can have the logic fail fast and skip certain lines